kernel/kernel-biometrics-api/src/main/java/io/mosip/kernel/biometrics/commons/BiometricsSignatureValidator.java

```diff
@@ -0,0 +1,142 @@
+import org.springframework.core.env.Environment;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.mosip.kernel.biometrics.entities.BIR;
+import io.mosip.kernel.biometrics.entities.BiometricRecord;
+import io.mosip.kernel.biometrics.entities.Entry;
+import io.mosip.kernel.biometrics.model.JWTSignatureVerifyRequestDto;
+import io.mosip.kernel.biometrics.model.JWTSignatureVerifyResponseDto;
+import io.mosip.kernel.core.exception.BaseCheckedException;
+import io.mosip.kernel.core.exception.ServiceError;
+import io.mosip.kernel.core.http.RequestWrapper;
+import io.mosip.kernel.core.http.ResponseWrapper;
+import io.mosip.kernel.core.util.CryptoUtil;
+import io.mosip.kernel.core.util.DateUtils;
+
+@Component
+public class BiometricsSignatureValidator {
+
+	private static final String DATETIME_PATTERN = "mosip.registration.processor.datetime.pattern";
+
+	public static final String TRUST_VALID = "TRUST_CERT_PATH_VALID";
+
+	@Autowired
+	private Environment env;
+
+	@Autowired
+	ObjectMapper mapper;
+
+	@Autowired
+	RestTemplate restTemplate;
+
+	@Value("${mosip.regproc.packet-validator.signature.disable-trust-validation:false}")
+	private Boolean disableTrustValidation;
+
+	public void validateSignature(BiometricRecord biometricRecord) throws JSONException, JsonParseException,
+			JsonMappingException, JsonProcessingException, IOException, BaseCheckedException {
+
+		List<BIR> birs = biometricRecord.getSegments();
+		for (BIR bir : birs) {
+			String sb = new String(bir.getSb(), StandardCharsets.UTF_8);
+			String bdb = Base64.getUrlEncoder().encodeToString(bir.getBdb());
+			if (bir.getOthers() != null) {
+				for (Entry entry : bir.getOthers()) {
+					if (entry.getKey().equals("PAYLOAD")) {
+						entry.setValue(entry.getValue().replace("<bioValue>", bdb));
+						String encodedPayloadValue = CryptoUtil.encodeBase64(entry.getValue().getBytes());
+						String constructedJWTToken = constructJWTToken(sb, encodedPayloadValue);
+						validateCertificateHeader(sb, new JSONObject(entry.getValue()));
+						validateJWTToken(constructedJWTToken);
+					}
+				}
+			}
+		}
+
+	}
+
+	private void validateCertificateHeader(String sb, JSONObject jsonObject) throws JSONException, BaseCheckedException {
+		String digitalIdHeader = new String(CryptoUtil.decodeBase64(jsonObject.getString("digitalId").split("\\.")[0]));
+		JSONObject digitalHeaderCertificate = new JSONObject(digitalIdHeader);
+		JSONArray digitalHeadercertificateJsonArray = digitalHeaderCertificate.getJSONArray("x5c");
+		String digitalCertificate = digitalHeadercertificateJsonArray.getString(0);
```

Comment:
If there are move than 1 certificate compare all the certs from jwt1 to jwt2

Reply:
matching all certificates.